---
title: "Class15"
author: "Evelyn Diaz - PID:A15576617"
date: "11/16/2021"
output: html_document
---

#Background
Our data for today came from Himes et al. RNASeq analysis of the drugdexamethasone, a synthetic glucocorticoid steroid with anti-inflammatory effects (Himes et al. 2014).

Read the countData and colData. 

```{r}
counts<- read.csv("airway_scaledcounts.csv", row.names = 1 )
metadata<- read.csv("airway_metadata (1).csv")

```


```{r}
head(counts)

```

```{r}
head(metadata)
```

>Q1. How many genes are in this dataset?

```{r}
nrow(counts)
```
There are 38694 genes in this data set. 


>Q2. How many ‘control’ cell lines do we have? 

```{r}
sum(metadata$dex =="control")

```
There are 4 control cell lines. 

>Q3. How would you make the above code in either approach more robust?
Like this:

FIrst, extract all the "control" columns. Then rake the rowwise mean to get the average count values for all genes in the four experiments. 
```{r}
control.inds<- metadata$dex == "control"
control.counts<- counts[, control.inds]
head(control.counts)
```

```{r}
control.mean<- rowMeans(control.counts)
```

>Q4. Follow the same procedure for the treated samples (i.e. calculate the mean per gene across drug treated samples and assign to a labeled vector called treated.mean)
Now for treated:

```{r}
treated.inds<- metadata$dex == "treated"
treated.counts<- counts[, treated.inds]
head(treated.counts)
treated.mean<- rowMeans(treated.counts)
```

We will combine our meancount data for bookkeeping purposes.

```{r}
meancounts <- data.frame(control.mean, treated.mean)

```


>Q5 (a). Create a scatter plot showing the mean of the treated samples against the mean of the control samples. Your plot should look something like the following.

Let's make a quick plot:
```{r}
plot(meancounts)
```

We need a log transformation to see details in the data.

We should plot on a log scale data
```{r}
plot(meancounts, log="xy")
```

we often use log2 because it has nice math properties that make interpretation easier.
```{r}
log2(10/10)

```

```{r}
log2(20/10)
```

```{r}
log2(5/10)
```
We see 0 values for no change and + values for increases and minus values for decreasing. This nice property leads us to work with **log2()fold change)** all the time in the genomics and proteomics. 

Let's add the **log2()fold change)** values to our `meancounts` dataframe.
```{r}
meancounts$log2fc <- log2(meancounts[,"treated.mean"]/meancounts[,"control.mean"])
head(meancounts)
```

I need to exclude the genes (rows) woth zaero counts because we can't say anythig about these as we have no data for them.

```{r}
head(meancounts[,1:2])
```

```{r}
head(meancounts[,1:2]== 0)

```

```{r}
which( c(F, F, T, T))
```
I can use the **which()** function with the `arr.ind=TRUE` argument to get the columns and rows where the TRUE values are (ie the zero counts in our case)

```{r}
zero.vals<- which(meancounts[,1:2]== 0, arr.ind = TRUE)
head(zero.vals)
```

```{r}
to.rm<- unique(zero.vals[, "row"])
head(sort(to.rm))
```

Now remove thee from out `meancounts` dataframe.

```{r}
mycounts <- meancounts[-to.rm,]
head(mycounts)
```

How many do we have left?

```{r}
nrow(mycounts)
```

Now we want to know how many upregulated and downregulated genes there are using a log2 threshold of +2 or - 2,respectively:

upregulated:
```{r}
sum(mycounts$log2fc > 2)
```

downregulated:
```{r}
sum(mycounts$log2fc< -2)
```

>Q10. Do you trust these results? Why or why not?
Yes I trust these results. 


#Section 4 DESeq2 analysis
Let’s do this the right way. DESeq2 is an R package specifically for analyzing count-based NGS data like RNA-seq. It is available from Bioconductor. 

```{r}
library(DESeq2)
```

We need to first set up the input.

```{r}
dds <- DESeqDataSetFromMatrix(countData=counts, 
                              colData=metadata, 
                              design=~dex)
dds
```

Now we can run DESeq analysis
```{r}
dds <- DESeq(dds)
```
To get at the results we use the deseq `results()` function:

```{r}
res <- results(dds)
res
```

#Section6 Data Visualization
#Volcano plots
Let’s make a commonly produced visualization from this data, namely a so-called Volcano plot. These summary figures are frequently used to highlight the proportion of genes that are both significantly regulated and display a high fold change.

```{r}
plot( res$log2FoldChange, res$padj)
```
That is not useful because all the small p-values are hidden at the bottom of hte plot and we can't see them. The log helps:

```{r}
plot( res$log2FoldChange, log(res$padj))
     
```

We can flip this pvalue axis by just putting a minus sign on it then we will have the classses volvano plot that the rest of the world uses. 

```{r}
plot( res$log2FoldChange, -log(res$padj))
```

Let's add some color to this plot to draw attention to the genes (ie points) we care about- that is those with large fold-change and low p-values ie (high-low pvalues)
```{r}
plot( res$log2FoldChange, -log(res$padj))
abline(v=c(-2, +2, col="gray"))
abline(h=-log(0.05), col="gray")
```
Now add some color:
```{r}
# Setup our custom point color vector 
mycols <- rep("gray", nrow(res))
mycols[ abs(res$log2FoldChange) > 2 ]  <- "red" 

inds <- (res$padj < 0.01) & (abs(res$log2FoldChange) > 2 )
mycols[ inds ] <- "blue"

# Volcano plot with custom colors 
plot( res$log2FoldChange,  -log(res$padj), 
 col=mycols, ylab="-Log(P-value)", xlab="Log2(FoldChange)" )

# Cut-off lines
abline(v=c(-2,2), col="gray", lty=2)
abline(h=-log(0.1), col="gray", lty=2)
```













